### PR TITLE
USB-LIB rework [rename usb-lib to usb_lib]

### PR DESCRIPTION
**Pull request by jvonau**

### CODE REVIEW

File: roles/usb_lib/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ini_file:
+    path: "{{ iiab_ini_file }}"
+    section: usb-lib
+    option: "{{ item.option }}"
+    value: "{{ item.value | string }}"
+  with_items:
+    - option: name
+      value: usb-lib
+    - option: description
+      value: '"usb-lib automounts Teacher Content on USB drives to /library/www/html/local_content, so students can browse it almost immediately at http://box/usb"'
```
Comment (holta) [inline roles/usb_lib/tasks/main.yml:26]:
```suggestion
      value: '"usb_lib automounts Teacher Content on USB drives to /library/www/html/local_content, so students can browse it almost immediately at http://box/usb"'
```

File: roles/4-server-options/tasks/main.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 - name: USB-LIB
```
Comment (holta) [inline roles/4-server-options/tasks/main.yml:25]:
```suggestion
- name: USB_LIB
```

File: roles/usb_lib/tasks/install.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# is owner correct under nginx?
+- name: Add dir {{ doc_root }}/local_content, where USB drive links can appear
+  file:
+    path: "{{ doc_root }}/local_content"
+    state: directory
+    owner: "{{ apache_user }}"
+    group: "{{ iiab_admin_user }}"    # ISN'T "{{ apache_user }}" MORE APPROPRIATE?
+    mode: 0775
+
+- name: 'Install from template: /etc/udev/rules.d/usbmount.rules, /etc/systemd/system/usbmount@.service, /usr/bin/iiab-usb-lib-show-all-on, /usr/bin/iiab-usb-lib-show-all-off'
```
Comment (holta) [inline roles/usb_lib/tasks/install.yml:10]:
```suggestion
- name: 'Install from template: /etc/udev/rules.d/usbmount.rules, /etc/systemd/system/usbmount@.service, /usr/bin/iiab-usb_lib-show-all-on, /usr/bin/iiab-usb_lib-show-all-off'
```

File: roles/usb_lib/tasks/install.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  template:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    mode: "{{ item.mode }}"
+  with_items:
+    - { src: 'usbmount@.service.j2' , dest: '/etc/systemd/system/usbmount@.service', mode: '0644' }
+    - { src: 'usbmount.rules.j2' , dest: '/etc/udev/rules.d/usbmount.rules', mode: '0644' }
+    - { src: 'iiab-usb-lib-show-all-on' , dest: '/usr/bin/', mode: '0755' }
```
Comment (holta) [inline roles/usb_lib/tasks/install.yml:18]:
```suggestion
    - { src: 'iiab-usb_lib-show-all-on' , dest: '/usr/bin/', mode: '0755' }
```

File: roles/usb_lib/tasks/install.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    - { src: 'iiab-usb-lib-show-all-off' , dest: '/usr/bin/', mode: '0755' }
```
Comment (holta) [inline roles/usb_lib/tasks/install.yml:19]:
```suggestion
    - { src: 'iiab-usb_lib-show-all-off' , dest: '/usr/bin/', mode: '0755' }
```

File: roles/usb_lib/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+- name: Install USB_LIB if 'usb_lib_installed' is not defined in {{ iiab_state_file }}
+  include_tasks: install.yml
+  when: usb_lib_install is undefined
+
+- name: Enable or Disable USB-LIB
```
Comment (holta) [inline roles/usb_lib/tasks/main.yml:5]:
```suggestion
- name: Enable or Disable USB_LIB
```

File: roles/usb_lib/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+- name: Enable/Disable/Restart Apache if primary
+  include_tasks: apache.yml
+  when: not nginx_enabled
+
+#- name: Enable/Disable/Restart NGINX if primary
+#  include_tasks: nginx.yml
+#  when: nginx_enabled | bool
+
+- name: Add 'usb-lib' variable values to {{ iiab_ini_file }}
```
Comment (holta) [inline roles/usb_lib/tasks/main.yml]:
```suggestion
- name: Add 'usb_lib' variable values to {{ iiab_ini_file }}
```

File: roles/usb_lib/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  ini_file:
+    path: "{{ iiab_ini_file }}"
+    section: usb-lib
```
Comment (holta) [inline roles/usb_lib/tasks/main.yml:19]:
```suggestion
    section: usb_lib
```

File: roles/usb_lib/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    option: "{{ item.option }}"
+    value: "{{ item.value | string }}"
+  with_items:
+    - option: name
+      value: usb-lib
```
Comment (holta) [inline roles/usb_lib/tasks/main.yml:24]:
```suggestion
      value: usb_lib
```

File: roles/usb_lib/tasks/main.yml (hunk 1 above)
Comment (holta) [inline roles/usb_lib/tasks/main.yml:26]:
```suggestion
      value: '"usb_lib automounts Teacher Content on USB drives to /library/www/html/local_content, so students can browse it almost immediately at http://box/usb"'
```
```suggestion
      value: '"usb-lib automounts Teacher Content on USB drives to /library/www/html/local_content, so students can browse it almost immediately at http://box/usb"'
```

**Review by holta — COMMENTED** [review]
1) Let's change from `usb-lib` to `usb_lib` across the board if at all possible.

2) Can this be a model for other `playbook-names-with-dashes` changing to `playbook_names_with_underscores` to reduce confusion/problems long term?

File: roles/usb_lib/tasks/main.yml (hunk 8 above)
Comment (jvonau) [reply roles/usb_lib/tasks/main.yml:19]: This will cause a dup section in iiab.ini so no

**Comment by holta** [comment]
Let's merge this after a fresh install on Raspberry Pi to be 100% sure it works there at minimum!

(Can someone test on Ubuntu as well?)

**Pull request closed without merging**